core/src/test/scala/unit/kafka/network/SocketServerTest.scala

```diff
@@ -179,15 +179,20 @@ class SocketServerTest extends JUnitSuite {
 
   @Test
   def testSocketsCloseOnShutdown() {
+    import java.util.concurrent.TimeUnit
     // open a connection
     val plainSocket = connect(protocol = SecurityProtocol.PLAINTEXT)
+    plainSocket.setTcpNoDelay(true)
     val traceSocket = connect(protocol = SecurityProtocol.TRACE)
+    traceSocket.setTcpNoDelay(true)
     val bytes = new Array[Byte](40)
     // send a request first to make sure the connection has been picked up by the socket server
     sendRequest(plainSocket, bytes, Some(0))
     sendRequest(traceSocket, bytes, Some(0))
     processRequest(server.requestChannel)
-
+    // Sleep for 200ms to not run into TCP SLOW_START behaviour potentially masking issues in
+    // sending data below
+    TimeUnit.MILLISECONDS.sleep(200L)
```

Comment:
@ijuma jup Linux and it seems it's a Mac vs. Linux thing. Tried Mac and get that nice `-1` ...
Same JDK version on Linux and Mac.

<img width="759" alt="screen shot 2017-03-04 at 15 05 11" src="https://cloud.githubusercontent.com/assets/6490959/23579357/178c0e5a-00ec-11e7-9bbe-91c89e9fa5b2.png">